envi/archs/amd64/opcode64.py

```diff
@@ -201,10 +198,10 @@
 ( 0, INS_MOV, ADDRMETH_G | OPTYPE_qp | OP_W, ADDRMETH_E | OPTYPE_ds | OP_R, ARG_NONE, cpu_AMD64, "movsxd", 0, 0, 0),
 ( 0, INSTR_PREFIX, 0, 0, 0, cpu_80386, 0, 0, 0, 0),  
 ( 0, INSTR_PREFIX, 0, 0, 0, cpu_80386, 0, 0, 0, 0),  
-(TBL_66, INSTR_PREFIX, 0, 0, 0, cpu_80386, 0, 0, 0, 0),  # 0x66
+(TBL_660F, INSTR_PREFIX, 0, 0, 0, cpu_80386, 0, 0, 0, 0),  # 0x66
```

Comment:
documentation on how the decoder works?

Reply:
Sort of. It does kind of skip past a few things, but technically isn't wrong in terms of output. While yes, it is supposed to be just 66, when we're ratcheting through the tables, if we're using 66 as an opcode byte, we know it leads immediately into an 0F.
So on one hand, it's a bit of a hack, but on the other, it saves us a full table rachet (and all that code entails) and it does work with the main disassembler loop to get other instructions decoding, so the trade off in my opinion is worth it. But it's not particularly a hill I'm willing to die on.

Reply:
And yes, I'm writing up docs on how the decoder works now.

Reply:
awesome.  thanks.  be sure to document it well, since it's a break from the intuitively pleasing paradigm.
is there a demonstrable performance benefit over 1 millon or 10 million decodes?

Reply:
A couples runs of a amd64 Performance on master (~5 million instructions decoded in a hot loop, so not entirely characteristic of performance, but as a rough estimate):
Run 1:
```
Mean of decoding time: 0.000010
Median of decoding time: 0.000009
Std Deviation of decoding time 0.000005
```
Run 2:
```
Mean of decoding time: 0.000011
Median of decoding time: 0.000009
Std Deviation of decoding time 0.000006
```
Run 3:
```
Mean of decoding time: 0.000011
Median of decoding time: 0.000009
Std Deviation of decoding time 0.000007
```
Run 4:
```
Mean of decoding time: 0.000010
Median of decoding time: 0.000009
Std Deviation of decoding time 0.000004
```

Same test, but on my amd64 branch:
```
Mean of decoding time: 0.000012
Median of decoding time: 0.000010
Std Deviation of decoding time 0.000007
```
Run 2:
```
Mean of decoding time: 0.000012
Median of decoding time: 0.000010
Std Deviation of decoding time 0.000006
```
Run 3:
```
Mean of decoding time: 0.000010
Median of decoding time: 0.000009
Std Deviation of decoding time 0.000004
```
Run 4:
```
Mean of decoding time: 0.000011
Median of decoding time: 0.000010
Std Deviation of decoding time 0.000006
```

So a the distribution of performance is about the same. Overall performance is in the worst case, only ever so *slightly* degraded, in exchange for full amd64 support.

I'm going to see if I can bring those numbers down, but it's not as bad as initially feared.

Reply:
Super quick n' dirty script here: https://gist.github.com/rakuy0/250ece4529c7b0e7f716e4c61c390d05